packages/valory/skills/transaction_settlement_abci/rounds.py

```diff
@@ -197,20 +191,9 @@ def should_check_late_messages(self) -> bool:
     @property
     def late_arriving_tx_hashes(self) -> List[str]:
         """Get the late_arriving_tx_hashes."""
-        late_arriving_tx_hashes_unparsed = cast(
-            List[str], self.db.get_strict("late_arriving_tx_hashes")
-        )
-        late_arriving_tx_hashes_parsed = []
-        for unparsed_hash in late_arriving_tx_hashes_unparsed:
-            if len(unparsed_hash) % TX_HASH_LENGTH != 0:
-                # if we cannot parse the hashes, then the developer has serialized them incorrectly.
-                raise ABCIAppInternalError(
-                    f"Cannot parse late arriving hashes: {unparsed_hash}!"
-                )
-            parsed_hashes = textwrap.wrap(unparsed_hash, TX_HASH_LENGTH)
-            late_arriving_tx_hashes_parsed.extend(parsed_hashes)
```

Comment:
simply not checking length here anymore. If it is wrong the developer made a mistake, one that should be caught elsewhere and ideally also earlier. This business logic would then have to become part of the abstract_round_abci. It might be worth considering using a specific type, something wrapping serialized data alongside it's expected length of individual "blocks".

Reply:
Please open an issue for what you described.

Reply:
I implemented this in a very simple manner for now. When the payload redesign occurs this may be revisited.